### PR TITLE
Arm NEON thumb support

### PR DESCRIPTION
**Pull request by emesare**

### CODE REVIEW

**Comment by emesare** [comment]
Should be well formatted now, if anyone has opinions on the formatting of calls they can push their formatting on top of this branch.

File: arch/armv7/thumb2_disasm/il_thumb2.cpp [hunk 1]
```diff
@@ -1475,7 +1512,289 @@ bool GetLowLevelILForNEONInstruction(Architecture* arch, LowLevelILFunction& il,
...
+			il.AddInstruction(WriteILOperand(il, instr, 1, ReadILOperand(il, instr, 3)));
+		}
+		if (instr->format->operandCount == 3)
+		{
+			if (instr->format->operands[2].type == OPERAND_FORMAT_REG_FP)
+			{
+				// r3:r2 <- d12
+				il.SetRegisterSplit(4, ReadILOperand(il, instr, 0), ReadILOperand(il, instr, 1),
+				                    ReadILOperand(il, instr, 3));
+			} else
```
Comment (galenbwill) [inline arch/armv7/thumb2_disasm/il_thumb2.cpp]: `else` should be on its own line

File: arch/armv7/thumb2_disasm/il_thumb2.cpp [hunk 2]
```diff
@@ -1475,7 +1512,289 @@ bool GetLowLevelILForNEONInstruction(Architecture* arch, LowLevelILFunction& il,
...
+			il.AddInstruction(il.SetRegister(regSize, reg, il.Add(regSize, il.Register(regSize, reg), ReadILOperand(il, instr, 2))));
+		}
+		else
+		{
+			il.AddInstruction(WriteILOperand(il, instr, 0, il.Load(regSize, GetMemoryAddress(il, instr, 1, 4))));
+		}
+		}
+		break;
+	case armv7::ARMV7_VSTR:
+		{
```
Comment (galenbwill) [inline arch/armv7/thumb2_disasm/il_thumb2.cpp]: don't indent this `{`, or the closing `}`, but do move the `break;` inside the curly braces so it is indented

File: arch/armv7/thumb2_disasm/il_thumb2.cpp [hunk 3]
```diff
@@ -1475,7 +1512,289 @@ bool GetLowLevelILForNEONInstruction(Architecture* arch, LowLevelILFunction& il,
...
+				prefix = "s";
+			}
+		}
+		auto regSize = RegisterSizeFromPrefix(prefix);
+		unsigned int d = instr->fields[FIELD_d];
+		unsigned int inc = 1;
+		if(IS_FIELD_PRESENT(instr, FIELD_inc))
+			inc = instr->fields[FIELD_inc];
+		int regs = instr->fields[FIELD_regs];
+		for(int i=0; i<regs; ++i) {
```
Comment (galenbwill) [inline arch/armv7/thumb2_disasm/il_thumb2.cpp]: spaces around binary operators, `{` on its own line

File: arch/armv7/thumb2_disasm/il_thumb2.cpp [hunk 4]
```diff
@@ -1475,7 +1512,289 @@ bool GetLowLevelILForNEONInstruction(Architecture* arch, LowLevelILFunction& il,
...
+		const char* prefix = "d";
+		if (IS_FIELD_PRESENT(instr, FIELD_single_regs)){
+			if (instr->fields[FIELD_single_regs] == 1) {
+				prefix = "s";
+			}
+		}
+		auto regSize = RegisterSizeFromPrefix(prefix);
+		unsigned int d = instr->fields[FIELD_d];
+		unsigned int inc = 1;
+		if(IS_FIELD_PRESENT(instr, FIELD_inc))
```
Comment (galenbwill) [inline arch/armv7/thumb2_disasm/il_thumb2.cpp]: space after `if`

File: arch/armv7/thumb2_disasm/il_thumb2.cpp [hunk 5]
```diff
@@ -1475,7 +1512,289 @@ bool GetLowLevelILForNEONInstruction(Architecture* arch, LowLevelILFunction& il,
...
+			int regIdx = (d + i * inc) % 32;
+			il.AddInstruction(il.Push(regSize, il.Register(regSize, GetRegisterByIndex(regIdx, prefix))));
+		}
+		}
+		break;
+	case armv7::ARMV7_VPOP:
+		{
+		// TODO: Clean this code up...
+		const char* prefix = "d";
+		if (IS_FIELD_PRESENT(instr, FIELD_single_regs)){
```
Comment (galenbwill) [inline arch/armv7/thumb2_disasm/il_thumb2.cpp]: newline before `{`

**Pull request merged**